src/Sarif/PropertiesDictionaryExtensionMethods.cs

```diff
@@ -119,6 +115,19 @@ public static class PropertiesDictionaryExtensionMethods
             writer.WriteEndElement(); // Properties
         }
 
+        private static void SaveSettingNameToDescriptionMap(XmlWriter writer, string key, ConcurrentDictionary<string, string> settingNameToDescriptionMap)
```

Comment:
>SaveSettingNameToDescriptionMap [](http://example.com/codeflow?start=28&length=31)

What are we doing here? Seems like you found a different issue and are oportunistically fixing it?

Reply:
there were no tests for this and when I add tests this issue show up and blocking my test pass so I should fix it together.

avoid injecting four calls to your new method ---- agree, let me take a look

Reply:
fixed to only one copy. the method is not needed.